core/services/pipeline/task_params.go

```diff
@@ -226,7 +227,12 @@ func (b *BytesParam) UnmarshalPipelineParam(val interface{}) error {
 			*b = BytesParam(bs)
 			return nil
 		}
-		*b = BytesParam(v)
+		// try decoding as base64 first, in case this is a string from the database
```

Comment:
Not fully happy with this, but the only alternative I can think of is pre-processing the `inputs`/`result` values in the ORM to wrap all nested `[]byte` values with a custom type that uses the `0x...` format (which BytesParam already decodes as hex values)

Reply:
@begmaroman was also running into this issue - did you two standardise on this fix?

Reply:
It's related but both fixes are needed. #4923 will solve the issue for toplevel `[]byte` values, i.e. `JSONSerializable {Value: []byte{'a'}}` but not for nested keys: `JSONSerializable {Value: map[string]interface{} {"a": []byte{'a'}}`